desc/objectives/_omnigenity.py

```diff
@@ -540,6 +541,319 @@ def compute(self, params, constants=None):
         return data["f_T"]
 
 
+class Omnigenity(_Objective):
+    """Omnigenity error.
+
+    Errors are relative to a target field that is perfectly omnigenous,
+    and are computed on a collocation grid in (ρ,η,α) coordinates.
+
+    This objective assumes that the collocation point (θ=0,ζ=0) lies on the contour of
+    maximum field strength |B|=B_max.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium to be optimized to satisfy the Objective.
+    field : OmnigenousField
+        Omnigenous magnetic field to be optimized to satisfy the Objective.
+    target : float, ndarray, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        len(target) must be equal to Objective.dim_f
+    bounds : tuple, optional
+        Lower and upper bounds on the objective. Overrides target.
+        len(bounds[0]) and len(bounds[1]) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+       Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    eq_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at for equilibrium data.
+        Must be a single flux suface.
+    field_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at for omnigenous field data.
+        Must be a single flux suface.
+    M_booz : int, optional
+        Poloidal resolution of Boozer transformation. Default = 2 * eq.M.
+    N_booz : int, optional
+        Toroidal resolution of Boozer transformation. Default = 2 * eq.N.
+    well_weight : float, optional
+        Weight applied to the bottom of the magnetic well (B_min) relative to the top
```

Comment:
what does this mean? if this is large, does it just try to match the minimum of the well and not care about the max?

Reply:
See updated documentation. But yes, if this is large it will prioritize the minimum of the well over the maximum. The theory is that deeply trapped particles are more problematic than shallowly trapped particles.